packages/components/src/ButtonWithIcon.tsx

```diff
@@ -7,15 +7,47 @@ export type ButtonProps = {
     e?: MouseEvent<HTMLButtonElement, globalThis.MouseEvent>
   ) => void | Promise<void> | null;
   testId?: string;
+  disabled?: boolean;
+  loading?: boolean;
 };
 
-export const ButtonWithIcon = ({ label, Icon, onClick, testId }: ButtonProps): ReactElement => {
+export const ButtonWithIcon = ({
+  label,
+  Icon,
+  onClick,
+  testId,
+  disabled,
+  loading,
+}: ButtonProps): ReactElement => {
+  const isLoading = loading !== undefined ? loading : false;
+  const isDisabled = isLoading ? true : disabled !== undefined ? disabled : false;
+
+  const disabledClasses = isDisabled ? 'cursor-not-allowed text-gray-600 bg-elevation' : '';
+
+  const IconOfState = (): ReactElement => {
+    if (isLoading) {
+      return (
+        <i className="pi pi-spinner animate-spin mt-1.5 mr-1.5" data-testid={'icon-loading'} />
+      );
+    }
+    return (
+      <>
+        {isDisabled ? (
+          <i className="pi pi-ban mt-1.5 mr-1.5" data-testid={'icon-disabled'} />
```

Comment:
I have not seen this pattern with the empty `<i>` before. Is it recommended by tailwind or somewhere else?